internal/terraform/rootmodule/root_module.go

```diff
@@ -138,81 +219,66 @@ func (rm *rootModule) initTfExecutor(dir string) (*exec.Executor, error) {
 		tf.SetTimeout(rm.tfExecTimeout)
 	}
 
-	return tf, nil
-}
+	rm.tfExec = tf
 
-func (rm *rootModule) initPluginCache(dir string) error {
-	var lf File
-	if rm.ignorePluginCache {
-		lf = &file{
-			path: pluginLockFilePaths(dir)[0],
-		}
-	} else {
-		var err error
-		lockPaths := pluginLockFilePaths(dir)
-		lf, err = findFile(lockPaths)
-		if err != nil {
-			if os.IsNotExist(err) {
-				rm.logger.Printf("no plugin cache found: %s", err.Error())
-				return nil
-			}
+	return nil
+}
 
-			return fmt.Errorf("unable to calculate hash: %w", err)
-		}
+func (rm *rootModule) discoverTerraformVersion(ctx context.Context) error {
+	if rm.tfExec == nil {
+		return nil
```

Comment:
might be more appropriate to return error since you have attempted to call `discoverTerraformVersion` before some critical setup func

Reply:
Yeah, I suppose it could return an error, but all errors from `load()` are just logged silently anyway, given the asynchronous nature of the loading.

More importantly it's not clear to me yet how, when and whether surface these errors to the user, so I created https://github.com/hashicorp/terraform-ls/issues/218 to revisit that problem later.